### ISSUE TITLE
Whisper model returns empty logits array

### ISSUE THREAD

**Issue by MahmoudAshraf97** [OpenNMT/CTranslate2#1822]
Hello, when enabling `return_logits_vocab` in whisper, an empty array is returned, this is reproducible with any input
`result[0].logits[0][0]`:
```
{'shape': (), 'typestr': '<f4', 'data': (0, False), 'version': 3}
```
Using CT2 4.5.0
